tests/unit/tracing_test.cpp

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/test/tracing.h>
+}
+
+extern bool *traced_syscalls;
+extern char *traced_syscalls_str;
+
+class tracing_mock : public KernelSingleton {
+public:
+   MOCK_METHOD(int, handle_sys_trace_arg, (const char *arg), (override));
+};
+
+TEST(simple_wildcard_match, basic)
+{
+   const char *str1 = "test string";
+   const char *str1_equivalent = "test string*";
+   const char *qmark_str1 = "tes? string";
+   const char *wrong_str1 = "test string1";
+   const char *asterisk_str1 = "test *";
+   const char *str2 = "test strin1";
+   const char *invalid_str = "te*t ";
+
+   ASSERT_EQ(true, simple_wildcard_match(str1, str1));
```

Comment:
Please, use ASSERT_TRUE(). Also, as you can see yourself, you're testing only for positive cases. That's not enough.

Reply:
I used ASSERT_TRUE and ASSERT_FALSE. But i will probably write more test cases (doesn't hurt)